sdk/servicebus/service-bus/src/util/atomXmlHelper.ts

```diff
@@ -44,51 +46,65 @@ export async function executeAtomXmlOperation(
+    if (webResource.method == "PUT") {
+      webResource.headers.set("content-length", Buffer.byteLength(webResource.body));
+    }
 
-  const reqPrepareOptions: RequestPrepareOptions = {
-    ...webResource,
-    headers: operationOptions.requestOptions?.customHeaders,
-    onUploadProgress: operationOptions.requestOptions?.onUploadProgress,
-    onDownloadProgress: operationOptions.requestOptions?.onDownloadProgress,
-    abortSignal: operationOptions.abortSignal,
-    spanOptions: operationOptions.tracingOptions?.spanOptions,
-    disableJsonStringifyOnBody: true
-  };
-  webResource = webResource.prepare(reqPrepareOptions);
-  webResource.timeout = operationOptions.requestOptions?.timeout || 0;
-  const response: HttpOperationResponse = await serviceBusAtomManagementClient.sendRequest(
-    webResource
-  );
+    log.httpAtomXml(`Executing ATOM based HTTP request: ${webResource.body}`);
+
+    const reqPrepareOptions: RequestPrepareOptions = {
+      ...webResource,
+      headers: operationOptions.requestOptions?.customHeaders,
+      onUploadProgress: operationOptions.requestOptions?.onUploadProgress,
+      onDownloadProgress: operationOptions.requestOptions?.onDownloadProgress,
+      abortSignal: operationOptions.abortSignal,
+      spanOptions,
+      disableJsonStringifyOnBody: true
+    };
+    webResource = webResource.prepare(reqPrepareOptions);
+    webResource.timeout = operationOptions.requestOptions?.timeout || 0;
+    const response: HttpOperationResponse = await serviceBusAtomManagementClient.sendRequest(
+      webResource
+    );
 
-  log.httpAtomXml(`Received ATOM based HTTP response: ${response.bodyAsText}`);
+    log.httpAtomXml(`Received ATOM based HTTP response: ${response.bodyAsText}`);
 
-  try {
-    if (response.bodyAsText) {
-      response.parsedBody = await parseXML(response.bodyAsText, { includeRoot: true });
+    try {
+      if (response.bodyAsText) {
+        response.parsedBody = await parseXML(response.bodyAsText, { includeRoot: true });
+      }
+    } catch (err) {
+      const error = new RestError(
+        `Error occurred while parsing the response body - expected the service to return valid xml content.`,
+        RestError.PARSE_ERROR,
+        response.status,
+        stripRequest(response.request),
+        stripResponse(response)
+      );
+      log.warning("Error parsing response body from Service - %0", err);
+      throw error;
     }
-  } catch (err) {
-    const error = new RestError(
-      `Error occurred while parsing the response body - expected the service to return valid xml content.`,
-      RestError.PARSE_ERROR,
-      response.status,
-      stripRequest(response.request),
-      stripResponse(response)
-    );
-    log.warning("Error parsing response body from Service - %0", err);
-    throw error;
-  }
 
-  return serializer.deserialize(response);
+    return serializer.deserialize(response);
+  } catch (e) {
+    span.setStatus({
```

Comment:
Might be useful to differentiate some of these error codes. In AppConfig I do this:

```typescript
  static getCanonicalCode(err: Error) {
    if (Spanner.isRestError(err)) {
      switch (err.statusCode) {
        case 401:
          return CanonicalCode.PERMISSION_DENIED;
        case 404:
          return CanonicalCode.NOT_FOUND;
        case 412:
          return CanonicalCode.FAILED_PRECONDITION;
      }
    }

    return CanonicalCode.UNKNOWN;
  }

  static isRestError(err: Error): err is RestError {
    return err instanceof RestError;
  }
```

I know that @xirzec has seen this code (since he's probably had to modify it). Is this something we should be doing for tracing (checking out the status code and providing a differentiated CanonicalCode?)

Reply:
If you can provide a better CanonicalCode it will help Azure Monitor. We were also trying to figure out a standard way to provide a more useful code/string for errors across Azure services, but I don't think we settled on one.